pkg/cloudprovider/aws/instance.go

```diff
@@ -185,21 +201,26 @@ func (p *InstanceProvider) getOverrides(instanceTypeOptions []cloudprovider.Inst
 				continue
 			}
 			for _, subnet := range subnets {
-				if aws.StringValue(subnet.AvailabilityZone) == offering.Zone {
-					override := &ec2.FleetLaunchTemplateOverridesRequest{
-						InstanceType: aws.String(instanceType.Name()),
-						SubnetId:     subnet.SubnetId,
-					}
-					// Add a priority for spot requests since we are using the capacity-optimized-prioritized spot allocation strategy
-					// to reduce the likelihood of getting an excessively large instance type.
-					// instanceTypeOptions are sorted by vcpus and memory so this prioritizes smaller instance types.
-					if capacityType == v1alpha1.CapacityTypeSpot {
-						override.Priority = aws.Float64(float64(i))
-					}
-					overrides = append(overrides, override)
-					// FleetAPI cannot span subnets from the same AZ, so break after the first one.
-					break
+				subnetZone := aws.StringValue(subnet.AvailabilityZone)
+				if subnetZone != offering.Zone || !zones.Has(offering.Zone) {
```

Comment:
Does it make sense to put the check
```
if !zones.Has(offering.Zone) {
  continue
}
```
 on line 203?

Reply:
Ofc we need to still do the check  `if subnetZone != offering.Zone { continue }` here